### PR TITLE
feat(calendar): ✨  version 1

### PR DESCRIPTION
**Pull request by navin-moorthy**

### CODE REVIEW

File: src/calendar-v1/CalendarGrid.ts [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+        case "ArrowRight":
+          event.preventDefault();
+          focusNextDay();
+          break;
+        case "ArrowDown":
+          event.preventDefault();
+          focusNextWeek();
+          break;
+      }
+    };
```
Comment (anuraghazra) [inline src/calendar-v1/CalendarGrid.ts:100]: We can refactor this handler to

```js

    const onKeyDown = createOnKeyDown({
      preventDefault: true,
      onKeyDown: htmlOnKeyDown,
      keyMap: event => {
        const shift = event.shiftKey;
        return {
          " ": selectFocusedDate,
          Enter: selectFocusedDate,
          End: focusEndOfMonth,
          Home: focusStartOfMonth,
          ArrowLeft: focusPreviousDay,
          ArrowUp: focusPreviousWeek,
          ArrowRight: focusNextDay,
          ArrowDown: focusNextWeek,
          PageUp: () => {
            shift ? focusPreviousYear() : focusPreviousMonth();
          },
          PageDown: () => {
            shift ? focusNextYear() : focusNextMonth();
          },
        };
      },
    });
```

File: src/calendar-v1/CalendarNextYearButton.ts [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+      ...htmlProps,
+    };
+  },
+});
+
+export const CalendarNextYearButton = createComponent({
+  as: "button",
+  memo: true,
+  useHook: useCalendarNextYearButton,
+});
```
Comment (anuraghazra) [inline src/calendar-v1/CalendarNextYearButton.ts:36]: Hmmm 🤔 these button components are very similar, maybe we can refactor this.

File: src/calendar-v1/CalendarState.ts [hunk 3]
```diff
@@ -0,0 +1,213 @@
+import { useState } from "react";
```
Comment (anuraghazra) [inline src/calendar-v1/CalendarState.ts:1]: 🤔 let's attribute react-aria here?

File: src/calendar-v1/stories/Calendar.stories.tsx [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+            ></path>
+          </svg>
+        </CalendarNextYearButton>
+      </div>
+      <CalendarGrid {...state} as="table" className="dates">
+        <thead>
+          <tr>
+            <th scope="col" abbr="Sunday">
+              Su
+            </th>
```
Comment (anuraghazra) [inline src/calendar-v1/stories/Calendar.stories.tsx:98]: just adding a remainder here for improving the the screen reader announcing the weekdays.

File: src/calendar-v1/CalendarWeekTitle.ts [hunk 5]
```diff
@@ -0,0 +1,37 @@
+import { BoxHTMLProps, BoxOptions, useBox } from "reakit";
+import { createComponent, createHook } from "reakit-system";
+import { CalendarStateReturn } from "./CalendarState";
+
+import { CALENDAR_WEEK_TITLE_KEYS } from "./__keys";
```
Comment (anuraghazra) [inline src/calendar-v1/CalendarWeekTitle.ts:5]: sort imports & local imports on group.

File: src/calendar-v1/CalendarWeekTitle.ts [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+  CalendarWeekTitleOptions,
+  CalendarWeekTitleHTMLProps
+>({
+  name: "CalendarWeekTitle",
+  compose: useBox,
+  keys: CALENDAR_WEEK_TITLE_KEYS,
+
+  useProps({ dayIndex, weekDays }, htmlProps) {
+    return {
+      "aria-label": weekDays[dayIndex].title,
```
Comment (anuraghazra) [inline src/calendar-v1/CalendarWeekTitle.ts:27]: just in case.

```diff
- "aria-label": weekDays[dayIndex].title,
+ "aria-label": weekDays[dayIndex]?.title,
```

**Comment by anuraghazra** [comment]
LGTM!

File: src/calendar-v1/CalendarWeekTitle.ts [hunk 7]
```diff
@@ -24,7 +24,7 @@ export const useCalendarWeekTitle = createHook<
 
   useProps({ dayIndex, weekDays }, htmlProps) {
     return {
-      "aria-label": weekDays[dayIndex].title,
+      "aria-label": weekDays[dayIndex]?.title,
```
Comment (navin-moorthy) [inline src/calendar-v1/CalendarWeekTitle.ts:27]: Dang 💢 

I miss this everytime 🤦

**Pull request merged**